dotnet/src/Connectors/Connectors.Memory.SqlServer.Classic/SqlServerMemoryStore.cs

```diff
@@ -0,0 +1,240 @@
+    {
+        await this._dbClient.DeleteCollectionAsync(collectionName, cancellationToken).ConfigureAwait(false);
+    }
+
+    /// <inheritdoc/>
+    public async Task<MemoryRecord?> GetAsync(string collectionName, string key, bool withEmbedding = false, CancellationToken cancellationToken = default)
+    {
+        if (string.IsNullOrWhiteSpace(collectionName))
+        {
+            throw new ArgumentNullException(nameof(collectionName));
+        }
+
+        SqlServerMemoryEntry? entry = await this._dbClient.ReadAsync(collectionName, key, withEmbedding, cancellationToken).ConfigureAwait(false);
+
+        if (!entry.HasValue) { return null; }
+
+        return this.GetMemoryRecordFromEntry(entry.Value);
+    }
+
+    /// <inheritdoc/>
+    public async IAsyncEnumerable<MemoryRecord> GetBatchAsync(string collectionName, IEnumerable<string> keys, bool withEmbeddings = false,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        if (string.IsNullOrWhiteSpace(collectionName))
+        {
+            throw new ArgumentNullException(nameof(collectionName));
+        }
+
+        await foreach (SqlServerMemoryEntry entry in this._dbClient.ReadBatchAsync(collectionName, keys, withEmbeddings, cancellationToken).ConfigureAwait(false))
+        {
+            yield return this.GetMemoryRecordFromEntry(entry);
+        }
+    }
+
+    /// <inheritdoc/>
+    public async Task<(MemoryRecord, double)?> GetNearestMatchAsync(string collectionName, ReadOnlyMemory<float> embedding, double minRelevanceScore = 0, bool withEmbedding = false, CancellationToken cancellationToken = default)
+    {
+        var nearest = this.GetNearestMatchesAsync(
+                    collectionName: collectionName,
+                    embedding: embedding,
+                    limit: 1,
+                    minRelevanceScore: minRelevanceScore,
+                    withEmbeddings: withEmbedding,
+                    cancellationToken: cancellationToken)
+            .WithCancellation(cancellationToken);
+
+        await foreach (var item in nearest)
+        {
+            return item;
+        }
+
+        return null;
+    }
+
+    /// <inheritdoc/>
+    public async IAsyncEnumerable<(MemoryRecord, double)> GetNearestMatchesAsync(string collectionName, ReadOnlyMemory<float> embedding, int limit, double minRelevanceScore = 0, bool withEmbeddings = false, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        if (string.IsNullOrWhiteSpace(collectionName))
+        {
+            throw new ArgumentNullException(nameof(collectionName));
+        }
+
+        if (limit <= 0)
+        {
+            yield break;
+        }
+
+        IAsyncEnumerable<(SqlServerMemoryEntry, double)> results = this._dbClient.GetNearestMatchesAsync(
+            collectionName: collectionName,
+            embedding: JsonSerializer.Serialize(embedding.ToArray()),
```

Comment:
`ToArray` is not needed (wastes extra allocation). `JsonSerializer` can serialize `ROM<float>`.